### PR TITLE
[IMP] account_permanent_lock_move: add a hook to bypass lock date according fields updated.

### PR DESCRIPTION
**Pull request by adrienpeiffer**

### CODE REVIEW

File: account_permanent_lock_move/models/account_move.py [hunk 1]
```diff
@@ -31,11 +31,18 @@ def create(self, vals):
         result._check_lock_date()
         return result
 
+    @api.model
+    def _get_allowed_write_field(self):
```
Comment (benwillig) [inline account_permanent_lock_move/models/account_move.py]: Maybe this method could have a more explicit name.

File: account_permanent_lock_move/models/account_move.py [hunk 2]
```diff
@@ -31,11 +31,18 @@ def create(self, vals):
...
         # Add _check_lock_date for write of account.move,
         # as it is not done by default
-        self._check_lock_date()
+        allowed_write_field = self._get_allowed_write_field()
+        if not set(allowed_write_field).issuperset(vals.keys()):
+            self._check_lock_date()
         result = super(AccountMove, self).write(vals)
-        self._check_lock_date()
+        if not set(allowed_write_field).issuperset(vals.keys()):
+            self._check_lock_date()
```
Comment (benwillig) [inline account_permanent_lock_move/models/account_move.py]: This code is duplicated, maybe we could move it to a specific method which would call ```_get_allowed_write_field```.

**Review by benwillig — CHANGES_REQUESTED** [review]

**Comment by adrienpeiffer** [comment]
@benwillig Done :smile:

**Review by benwillig — APPROVED** [review]

**Review by feketemihai — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please change module version number.

**Pull request closed without merging**